packages/core/src/modules/credentials/CredentialsApi.ts

```diff
@@ -324,12 +325,19 @@ export class CredentialsApi<CPs extends CredentialProtocol[]> implements Credent
     return credentialRecord
   }
 
-  public async declineOffer(credentialRecordId: string): Promise<CredentialExchangeRecord> {
+  public async declineOffer(
+    credentialRecordId: string,
+    options?: DeclineCredentialOfferOptions
+  ): Promise<CredentialExchangeRecord> {
     const credentialRecord = await this.getById(credentialRecordId)
     credentialRecord.assertState(CredentialState.OfferReceived)
 
     // with version we can get the Service
     const protocol = this.getProtocol(credentialRecord.protocolVersion)
+    if (options?.sendProblemReport) {
+      await this.sendProblemReport({ credentialRecordId, description: 'Offer declined' })
```

Comment:
I think it would be nice to make it possible to specify the description (and/or the code), as this can be triggered for different reasons (user declined, VDR not supported, timeout, etc) and credential protocols are currently making this very 'open'. This comment of course also applies to the proofs module counterpart.

Reply:
I was thinking of it as well yes but was lazy 😅

I'll make the changes (for now with extra problemReportDescription param that is only used when you also send a problem report) later we can make it an object